cdk/ecs/lib/app.ts

```diff
@@ -0,0 +1,122 @@
+
+class ApplicationSignalsECSDemo {
+    private readonly app: App;
+    private adotJavaImageTag: string;
+    private adotPythonImageTag: string;
+    private ecsClusterStack: EcsClusterStack;
+    private serviceDiscoveryStack: ServiceDiscoveryStack;
+    private loadbalancerStack: LoadBalancerStack;
+    private logStack: LogStack;
+    private rdsDatabaseStack: RdsDatabaseStack;
+
+    constructor() {
+        this.app = new App();
+        this.adotJavaImageTag = '';
+        this.adotPythonImageTag = '';
+        this.runApp();
+    }
+
+    public async runApp(): Promise<void> {
+        const [adotJavaImageTag, adotPythonImageTag] = await Promise.all([
+            getLatestAdotJavaTag(),
+            getLatestAdotPythonTag(),
+        ]);
+
+        assert(adotJavaImageTag !== '', 'ADOT Java Image Tag is empty');
+        assert(adotPythonImageTag !== '', 'ADOT Python Image Tag is empty');
+
+        this.adotJavaImageTag = adotJavaImageTag;
+        this.adotPythonImageTag = adotPythonImageTag;
+
+        // Execute synchronous operations
+        this.createStacks();
+        this.createServers();
+        this.createApiGateway();
+        this.runServices();
+        this.generateTraffic();
+        this.app.synth();
+    }
+
+    private createStacks(): void {
+        const networkStack = new NetworkStack(this.app, 'NetworkStack');
+
+        this.logStack = new LogStack(this.app, 'LogStack');
+
+        this.loadbalancerStack = new LoadBalancerStack(this.app, 'LoadBalancerStack', {
+            vpc: networkStack.vpc,
+            securityGroup: networkStack.albSecurityGroup,
+        });
+
+        this.rdsDatabaseStack = new RdsDatabaseStack(this.app, 'RdsDatabaseStack', {
+            vpc: networkStack.vpc,
+            rdsSecurityGroup: networkStack.rdsSecurityGroup,
+        });
+
+        const iamRolesStack = new IamRolesStack(this.app, 'IamRolesStack');
+
+        // Grant ecsTaskRole access to database
+        this.rdsDatabaseStack.dbSecret.grantRead(iamRolesStack.ecsTaskRole);
+        this.rdsDatabaseStack.dbSecret.grantWrite(iamRolesStack.ecsTaskRole);
+
+        this.serviceDiscoveryStack = new ServiceDiscoveryStack(this.app, 'ServiceDiscoveryStack', {
+            vpc: networkStack.vpc,
+        });
+
+        this.ecsClusterStack = new EcsClusterStack(this.app, 'EcsClusterStack', {
+            vpc: networkStack.vpc,
+            securityGroups: [networkStack.ecsSecurityGroup],
+            ecsTaskRole: iamRolesStack.ecsTaskRole,
+            ecsTaskExecutionRole: iamRolesStack.ecsTaskExecutionRole,
+            subnets: networkStack.vpc.publicSubnets,
+            serviceDiscoveryStack: this.serviceDiscoveryStack,
+            logStack: this.logStack,
+        });
+    }
+
+    private createServers() {
+        this.ecsClusterStack.createConfigServer();
+        this.ecsClusterStack.createDiscoveryServer();
+        this.ecsClusterStack.createAdminServer();
+    }
+
+    private createApiGateway() {
```

Comment:
I would prefer not to use ApiGateway. It might confuse people that we are using Amazon API Gateway service. Suggest to use pet clinic frontend instead.